File: Leetcode/python/sol2788.py
# https://leetcode.com/problems/split-strings-by-separator/description/
class Solution:
    def splitWordsBySeparator(self, words: List[str], separator: str) -> List[str]:

        result = []
        for i in words:
            temp = i.split(separator)
            result += list(filter(lambda x: len(x ) >0, temp))

        return result

File: Leetcode/javascript/sol2788.js
/**
 * @param {string[]} words
 * @param {character} separator
 * @return {string[]}
 */
var splitWordsBySeparator = function(words, separator) {

    var result = [];

    for(let i of words){
        let temp = i.split(separator);
        temp = temp.filter(x => x.length > 0)
        result.push(...temp)
    }
    return result;
};